### ISSUE TITLE
Unable to compile compiler_builtins (missing absvdi2.c)

### ISSUE THREAD

**Issue by tones111** [redox-os/redox#875]
**Reproduction**: 
run make all from redox commit 404cec085c470f216a89d2516252978a9281ddf2

**Expected behavior**:
successful build

**Actual behavior**:
build errors during compilation of compiler_builtins with...
running: "/home/paul/src/redox/libc-artifacts/gcc.sh" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-fno-stack-protector" "-U_FORTIFY_SOURCE" "-m64" "-fno-builtin" "-fvisibility=hidden" "-fomit-frame-pointer" "-ffreestanding" "-DVISIBILITY_HIDDEN" "-o" "/home/paul/src/redox/rust/src/target/x86_64-unknown-redox/release/build/compiler_builtins-2b8e36ca5c56eee4/out/compiler-rt/../compiler-rt/lib/builtins/absvdi2.o" "-c" "../compiler-rt/lib/builtins/absvdi2.c"
cargo:warning=gcc: error: ../compiler-rt/lib/builtins/absvdi2.c: No such file or directory
ExitStatus(ExitStatus(256))

**Build information**:
$ rustc -V
rustc 1.17.0-nightly (0aeb9c129 2017-03-15)

$ git rev-parse HEAD
404cec085c470f216a89d2516252978a9281ddf2

$ uname -a
Linux mogul 4.10.2-1-ARCH #1 SMP PREEMPT Mon Mar 13 17:13:41 CET 2017 x86_64 GNU/Linux

**Comment by bjorn3** [comment]
Have you run `git submodule update --init` before?

**Comment by tones111** [comment]
I did.  submodules were up to date.  I had tried to build from the previous commit because my submodule update was failing (see https://github.com/redox-os/rust/pull/2).  I then fetched the missing commits from upstream and tried to build 404cec0 with the resulting error.

I tried git clean -dxf from the redox and rust repos and make clean, but the error was still there.  I then did an rm -rf on my redox repo, re-cloned with the redox/rust pull request having been merged and it's been able to get past the issue.

Is there something make clean could do to better clean out the toolchain?
